### ISSUE TITLE
Wrong tags parsing on Bandcamp

### ISSUE THREAD

**Issue by fizmatbfu** [web-scrobbler/web-scrobbler#4074]
Wrong parsing on page https://room40.bandcamp.com/album/audiodrome

Artist should be `Phauss`, on Scrobbler shows `Audiodrome (Zürich`
Track name should be `Audiodrome (Zürich - Zürich 1986)`, on Scrobbler shows `Zürich 1986)`

![wrongTags](https://github.com/web-scrobbler/web-scrobbler/assets/36421069/4e786aab-4b4e-46cb-a1d6-ee0db9180331)

Also some users scrobbled wrong artist name : https://www.last.fm/music/Audiodrome%20(Z%C3%BCrich

**Environment**
-   Windows 10
-   Mozilla Firefox 116.0.3
-   Extension version: 3.0.15

**Comment by yayuyokitano** [comment]
The reason this is happening is that the bandcamp connector has behavior for compilation releases (example: https://krefeld8ung.bandcamp.com/album/krefeld-8ung-vol-1).
It seems that since every track on the release has a hyphen, the connector is false positiving it as being a compilation, and getting artist and track both from the title.
I guess we could add an exception for when the hyphen is within parentheses or something?

**Comment by schosin** [comment]
Ran into the same issue. Workaround with parentheses would not work for the following case:

https://antrisch.bandcamp.com/album/expedition-ii-die-passage
https://www.last.fm/music/I%20FESTGEFROREN/_/Packeisfalle

In the collection page, the current title seems to be present in `[data-bind="text: currentTrack().trackTitle"]`. In my example, this includes the full title.

Interestingly enough, a compilation (https://bbemusic.bandcamp.com/album/j-jazz-vol-4-deep-modern-jazz-from-japan-nippon-columbia-1968-1981?from=hp, first result when searching for any compilation on bandcamp for me), when listening to this on someone elses collection page, only the title ("Exchange" will be rendered). Not sure if this will always work, but if we remove the suffix "Exchange" from the title "Takeo Moriyama - Exchange" (however we could access that) we could extract the artist name for this track.

**Comment by yayuyokitano** [comment]
Yeah there are some compilations where it is separated and possible to positively identify. However on some albums (like the one I linked) that does not seem to be a thing.
I'm not familiar enough with the website to say what is a more common issue than what though, these decisions were made by others long ago.

**Comment by jaccarmac** [comment]
My experience with Bandcamp is that compilations are more common than albums with dashes in every title but by one artist. However, while the current behavior makes sense that way, I understand it's annoying in this case (and in the case where a compilation isn't picked up because of a single track without a dash!). Perhaps it would be worth it to add some kind of notification for this case to make users more aware of the behavior? Babysitting Bandcamp scrobbles is the only solution I have come up with, myself.

As for the BBE record, I didn't go and test it myself, but I believe that playing compilation tracks from collection pages (or Bandcamp radio) always has weird behavior: The site doesn't surface the full album info, just the currently-playing track. (I'm pretty sure a similar issue exists for private albums in the app for Android scrobblers, even).

**Comment by ZoomRmc** [comment]
I've run into the same issue.

This hidden behaviour should be optional, possibly with a temporary suppression button for the current page.

Personally, I'm much less interested in proper scrobbling of compilation albums than normal records, so I'd just disable that if it were an option.

Another solution is to perform the substitution only if "VA"/"Various Artists" in the artist label or "Compilation"/"Split" in the album label are found, but this is probably too error-prone.
